dmscripts/generate_database_data.py

```diff
@@ -0,0 +1,48 @@
+import random
+import string
+import os
+from dmapiclient import DataAPIClient
+
+from typing import Optional
+
+USER_ROLES = [
+    'admin',
+    'admin-ccs-category',
+    'admin-ccs-sourcing',
+    'admin-manager',
+    'admin-framework-manager',
+    'admin-ccs-data-controller',
+    'buyer',
+    'supplier',
+]
+
+
+DEFAULT_PASSWORD = os.getenv("DM_DEFAULT_PASSWORD", default=None)
+
+
+def generate_email_address(username: Optional[int] = None) -> str:
+    """Generate an email of the form 123456@user.marketplace.team"""
+    if username is not None:
+        return f"{username}@user.marketplace.team"
+    else:
+        return f"{random.randrange(1, 10 ** 6):06}@user.marketplace.team"
+
+
+def random_string_of_length(n: int = 10) -> str:
+    return ''.join(random.choice(string.ascii_letters) for _ in range(n))
+
+
+def generate_user(data: DataAPIClient, role: Optional[str] = None) -> dict:
```

Comment:
What happens if the role isn't provided?

Reply:
Good catch - this is leftover from some earlier refactoring and the role should be required